File: python/0242.py
from collections import defaultdict
from typing import Dict

class Solution:
    def countLetters(self, s: str) -> Dict[str, int]:
        letter_counts = defaultdict(int)
        for letter in s:
            letter_counts[letter] += 1
        return letter_counts

    def isAnagram(self, s: str, t: str) -> bool:
        letters_s = self.countLetters(s)
        letters_t = self.countLetters(t)
        return letters_s == letters_t            

File: go/0242.go
package leetcode

func countLetters(word string) [26]int {
	letterToFrequency := [26]int{}
	for _, letter := range word {
		letterToFrequency[int(letter-'a')] += 1
	}
	return letterToFrequency
}

func isAnagram(s string, t string) bool {
	letterCountsT := countLetters(t)
	letterCountsS := countLetters(s)

	for idx, freq := range letterCountsS {
		if letterCountsT[idx] != freq {
			return false
		}
	}
	return true
}
